warehouse/packaging/models.py

```diff
@@ -603,6 +603,17 @@ def has_meta(self):
             ]
         )
 
+    @property
+    def trusted_published(self) -> bool:
+        """
+        A Release can be considered published via a trusted publisher if
+        **all** the Files in the release are published via a trusted publisher.
+        """
+        files = self.files.all()  # type: ignore[attr-defined]
```

Comment:
Since `self.files` returns an `AppenderQuery`, it returns True until we resolve the query by fetching the rows. If there's a better way to do this, I'm all ears!